File: exercism/python/resistor-color-duo/resistor_color_duo.py
def color_code(color: str) -> int:
    """
    Translate resistor color to resistance value
    @type color: resistor color
    @return int: resistance value
    """
    if color == "black":
        return 0
    elif color == "brown":
        return 1
    elif color == "red":
        return 2
    elif color == "orange":
        return 3
    elif color == "yellow":
        return 4
    elif color == "green":
        return 5
    elif color == "blue":
        return 6
    elif color == "violet":
        return 7
    elif color == "grey":
        return 8
    elif color == "white":
        return 9


def value(colors: list[str]) -> int:
    translated = []
    for color in colors:
        translated.append(color_code(color))
    return translated[0:2][0] * 10 + translated[0:2][1]

File: exercism/typescript/resistor-color-duo/resistor-color-duo.ts
export const COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];

export function decodedValue(bandsColors: string[]): number {
  let resistance: string = "";
  for (let i = 0; i <= bandsColors.length && i < 2; i++) {
    resistance += "" + COLORS.indexOf(bandsColors[i].toLowerCase());
  }
  return parseInt(resistance);
}
